src/main/java/baseball/Application.java

```diff
@@ -1,7 +1,105 @@
+import static baseball.util.PrintUtils.println;
+import static baseball.util.ValidationUtils.validateInput;
+
+import baseball.dto.Score;
+import camp.nextstep.edu.missionutils.Console;
+import camp.nextstep.edu.missionutils.Randoms;
+import java.util.ArrayList;
+import java.util.List;
+
 public class Application {
+    private static boolean isGameToBeContinued = true;
+
     public static void main(String[] args) {
-        // TODO: 프로그램 구현
+        playGame();
+    }
+
+    private static void playGame() {
+        printMessage(START_MESSAGE);
+        do {
+            List<Integer> pickedIntegers = getRandomIntegers();
+
+            while (true) {
+                printMessage(INPUT_NUMBER_PROMPT);
+
+                String inputValue = getInputValue();
+                Score score = getScoreFrom(inputValue, pickedIntegers);
+
+                printScoreBoardMessageFrom(score);
+                printIfStrikeOutFrom(score);
+
+                if (score.isStrikeOut()) {
+                    break;
+                }
+            }
+            askIfContinue();
+        } while (isGameToBeContinued);
+        Console.close();
+    }
+
+    private static Score getScoreFrom(String inputValue, List<Integer> pickedIntegers) {
+        Score score = Score.builder().build();
+
+        for (int i = 0; i < inputValue.length(); i++) {
+            for (int j = 0; j < pickedIntegers.size(); j++) {
+                if (inputValue.charAt(i) == pickedIntegers.get(j).toString().charAt(0) && i == j) {
+                    score.increaseStrikeCount();
+                    break;
+                }
+
+                if (inputValue.charAt(i) == pickedIntegers.get(j).toString().charAt(0)) {
+                    score.increaseBallCount();
+                }
+            }
+        }
+        return score;
+    }
+
+    private static String getInputValue() {
+        String inputValue = Console.readLine();
+
+        validateInput(inputValue);
+        return inputValue;
+    }
+
+    private static List<Integer> getRandomIntegers() {
+        List<Integer> pickedIntegers = new ArrayList<>();
+        while (pickedIntegers.size() < 3) {
+            int pickedInteger = Randoms.pickNumberInRange(1, 9);
+            if (!pickedIntegers.contains(pickedInteger)) {
+                pickedIntegers.add(pickedInteger);
+            }
+        }
+
+        return pickedIntegers;
+    }
+
+    private static void printScoreBoardMessageFrom(Score score) {
+        if (score.isNothing()) {
+            printMessage(NOTHING);
+            return;
+        }
+        println(score.getScoreBoardMessage());
+    }
+
+    private static void printIfStrikeOutFrom(Score score) {
+        if (score.isStrikeOut()) {
+            printMessage(END_MESSAGE);
+        }
+    }
+
+    private static void askIfContinue() {
+        printMessage(GAME_END_PROMPT);
+
+        isGameToBeContinued = !Console.readLine().equals(FINISH.getNumber());
```

Comment:
한 라인에 점(`.`)이 많으면 잘 짜고 있는지 의심해볼 필요가 있는 것 같아요!
디미터 법칙을 찾아 보면 좋을 것 같습니다!!

Reply:
`.`을 최대한 많이 줄였다고 생각했는데 이 부분도 좀 더 줄였어야 됐네요. 

GameOverSignal로 동등성 판단 로직을 옮기고 콘솔 라이브러리도 스태틱 임포트로 바꿔 보면 괜찮지 않을까 싶습니다.
```suggestion
        isGameToBeContinued = GameOverSignal.isNotFinished(readLine());
```